docs/arcaflow/workflows/expressions.md

```diff
@@ -28,31 +85,274 @@ $.foo.bar
 
+To use a map in an expression, the expression to the left of the brackets must be a reference to a map. That is then followed by a pair of brackets with a sub-expression between them. That sub-expression must evaluate to a valid key in the map.
+
+Here is an example of a map whose key are strings, and whose values are integers. The map is stored in a field called `foo` in the root-level object:
+
+```yml
+foo:
+  a: 1
+  b: 2
 ```
-$.foo.0
+
+Given the map shown above, the following expression would yield a value of `2`:
+```JavaScript
+$.foo["b"]
 ```
 
-## Array accessor
+## Functions
 
-Alternative to the dot notation, you can also use the array accessor:
+Functions are built-in tasks with pre-defined behavior and a known input and output schema.
 
+Functions are defined by the engine.
+
+Functions:
+TO BE DEFINED BEFORE MERGE.
+
+The syntax for a function has multiple parts. First, you have the function's identifying name, followed by `(`, followed by 0 or more comma-separated expressions, followed by `)`.
+
+Example:
+```JavaScript
+thisIsAFunction("this is a string literal for the first parameter", $.a.b)
 ```
-$.foo[0]
-```
 
-In Arcaflow, the two are equivalent.
+## Binary Operations
+
+Binary Operations have an expression to the left and right, with an operator in between.
+The order of operations determines which operators run first. See [Order of Operations](#order-of-operations)
+
+| Operator| Description        |
+| --------|--------------------|
+| `+`     | [Addition/Concatenation](#additionconcatenation) |
+| `-`     | [Subtraction](#subtraction) |
+| `*`     | [Multiplication](#multiplication)|
+| `/`     | [Division](#division)|
+| `%`     | [Modulus](#modulus)|
+| `^`     | [Exponentiation](#exponentiation)|
+| `==`    | [Equal To](#equal-to)|
+| `!=`    | [Not Equal To](#not-equal-to)|
+| `>`     | [Greater Than](#greater-than)|
+| `<`     | [Less Than](#less-than)|
+| `>=`    | [Greater Than or Equal To](#greater-than-or-equal-to)|
+| `<=`    | [Less Than or Equal To](#less-than-or-equal-to)|
+| `&&`    | [Logical And](#logical-and)|
+| `\|\|`  | [Logical Or](#logical-or)|
+
+
+### Addition/Concatenation
+
+This operator has different behavior depending on the type.
+
+##### String Concatenation
+
+When the `+` operator is used with two strings, it concatenates them together.
+For example, the expression `"a" + "b"` would output the string `"ab"`.
+
+##### Mathematical Addition
+
+When the `+` operator is used with numerical operands, it adds them together. The operator requires numerical operands with the same type. You cannot mix float and integer operands.
+For example, the expression `2 + 2` would output the integer `4`.
+
+
+### Subtraction
+
+When the `-` operator is used with a numerical operands, subtracts the value of the right operand from the value of the left. The operator requires numerical operands with the same type. You cannot mix float and integer operands.
```

Comment:
"subtracts" feels lonely, and you might follow the indefinite prepositional model of addition by claiming that "it" (the operator) subtracts the value [...].